backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -628,45 +628,120 @@ public void validateDiffMergeRequest(
-      List<CommitEntity> parentCommits = Arrays.asList(internalCommitA, internalCommitB);
-      String mergeMessage = request.getContent().getMessage();
-      if (mergeMessage.isEmpty()) {
-        mergeMessage =
-            "Merge "
-                + request.getCommitShaA().substring(0, 7)
-                + " into "
-                + request.getCommitShaB().substring(0, 7);
-      }
-      UserInfo currentLoginUserInfo = authService.getCurrentLoginUserInfo();
-      String author = authService.getVertaIdFromUserInfo(currentLoginUserInfo);
-      final String commitSha =
-          VersioningUtils.generateCommitSHA(parentSHAs, mergeMessage, timeCreated, author, rootSha);
-
-      Commit internalCommit =
-          Commit.newBuilder()
-              .setDateCreated(timeCreated)
-              .setAuthor(author)
-              .setMessage(mergeMessage)
-              .setCommitSha(commitSha)
-              .build();
-      RepositoryEntity repositoryEntity =
-          repositoryDAO.getRepositoryById(writeSession, request.getRepositoryId());
-      CommitEntity commitEntity =
-          new CommitEntity(repositoryEntity, parentCommits, internalCommit, rootSha);
-      writeSession.saveOrUpdate(commitEntity);
-      writeSession.getTransaction().commit();
-      return MergeRepositoryCommitsRequest.Response.newBuilder()
-          .setCommit(commitEntity.toCommitProto())
-          .build();
+              locationBlobsMapCommitASimple,
+              conflictLocationMap);
+
+      if (conflictLocationMap.isEmpty()) {
+        final String rootSha = setBlobs(writeSession, blobContainerList, new FileHasher());
+        long timeCreated = new Date().getTime();
+        List<String> parentSHAs = Arrays.asList(request.getCommitShaA(), request.getCommitShaB());
+        List<CommitEntity> parentCommits = Arrays.asList(internalCommitA, internalCommitB);
+        String mergeMessage = request.getContent().getMessage();
+        if (mergeMessage.isEmpty()) {
+          mergeMessage =
+              "Merge "
+                  + request.getCommitShaA().substring(0, 7)
+                  + " into "
+                  + request.getCommitShaB().substring(0, 7);
+        }
+        UserInfo currentLoginUserInfo = authService.getCurrentLoginUserInfo();
+        String author = authService.getVertaIdFromUserInfo(currentLoginUserInfo);
+        final String commitSha =
+            VersioningUtils.generateCommitSHA(
+                parentSHAs, mergeMessage, timeCreated, author, rootSha);
+
+        Commit internalCommit =
+            Commit.newBuilder()
+                .setDateCreated(timeCreated)
+                .setAuthor(author)
+                .setMessage(mergeMessage)
+                .setCommitSha(commitSha)
+                .build();
+        CommitEntity commitEntity =
+            new CommitEntity(
+                repositoryFunction.apply(writeSession), parentCommits, internalCommit, rootSha);
```

Comment:
in master there is no repositoryFunction in this method. You should call repositoryDAO.RepositoryEntity getRepositoryById(Session session, RepositoryIdentification id, boolean checkWrite=true) instead

Reply:
fixed